nextjs-with-firebase-hosting/README.md

```diff
@@ -0,0 +1,56 @@
+# Next.js with Firebase Hosting Rewrites
+
+Host your SSR Next.js app on Cloud Functions with Firebase Hosting rewrite rules. Using a rewrite rule that catches **ALL** routes we can then host our SSR app on our Firebase Hosting URL instead of the Firebase Cloud Function URL.
+
+## How to use
+
+```bash
+git clone https://github.com/firebase/functions-samples
+cd nextjs-with-firebase-hosting
+yarn install
+```
+
+It is recommended to use a package manager that uses a lockfile and caching for faster dev/test cycles:
+- [Yarn](https://github.com/yarnpkg/yarn)
+- [npm5.1.x](https://github.com/npm/npm)
+- [pnpm](https://github.com/pnpm/pnpm)
+
+Set up firebase:
+- create a project through the [firebase web console](https://console.firebase.google.com/)
+- grab the projects ID from the web consoles URL: https://console.firebase.google.com/project/<projectId>
+- update the `.firebaserc` default project ID to the newly created project
```

Comment:
This repo's `.gitignore` ignores that file type unfortunately. I could update the instructions here to inform the user of the need to create one? Should that change be made on this branch? (I'm new to OSS and not sure of the protocol on updating a merged branch)